articles/parametric-vs-nonparametric/index.md

```diff
@@ -0,0 +1,123 @@
+
+[K-nearest neighbors](https://www.geeksforgeeks.org/k-nearest-neighbours/) is an example of a non-parametric algorithm. Its predictions are founded on the k most similar training patterns for a new instance of data. We should note that there are no assumptions made about the form of the mapping function aside from one. It is assumed that most alike' training patterns have a high likelihood of producing a similar output.
+
+We mentioned that linear SVM is an example of a parametric model. This is because basic support vector machines are linear classifiers. However, SVMs not constrained by a set number of parameters are considered non-parametric.
+
+### Parametric vs Non-Parametric models
+
+Since we can now define both parametric and non-parametric models, we compare both in this section. We then look at the benefits and limitations of both types of models.
+
+
+![fit](/engineering-education/parametric-vs-nonparametric/fit.png)
+
+*Results of parametric and non-parametric regression.*
+
+[Source](https://medium.com/analytics-vidhya/parametric-and-nonparametric-models-in-machine-learning-a9f63999e233)
+
+Let’s first note that the data points in the two scenarios in the image above are the same.
+
+Recalling the equation that we described earlier, the first image illustrates the mapping function as a linear regression line. This represents a parametric model. We see the consequence of the linear function on the data. A lot of data points are ignored.
+
+The image with a wiggly function represents a non-parametric model. As we mentioned, these algorithms make little to no guesses about the mapping function. As a result, they show great flexibility and offer a better fit to data than parametric ones.
+
+#### Benefits
+
+##### Parametric Models
+
+**Simplicity.** The methods of parametric algorithms are easier to understand. The interpretability of results is also easier in comparison to non-parametric models.
+
+**Training data.** Parametric algorithms require less training data than non-parametric ones.
+
+**Training speed.** They are computationally faster than non-parametric methods. They can be trained faster than non-parametric ones since they usually have fewer parameters to train.
+
+##### Non-Parametric Models
+
+**Performance.** Non-parametric models may offer more accurate predictions since they offer a better fit to data than parametric ones.
+
+**Flexibility.** As shown by the image above, these algorithms provide a good fit for data. They can fit many forms of a function.
+
+**Little to no assumptions.** Little to no guesses about the mapping function is made. Compared to parametric algorithms, non-parametric algorithms learn more from data. This is because the learning of parametric algorithms may be limited by the assumptions they make.
+
+#### Limitations
+
+##### Parametric Models
```

Comment:
Can you talk about storing these models? This angle is necessary to understand parametric and non-parametric models.